### PR TITLE
Fix unclaimed redeemers list

### LINKED ISSUE #206: accountsWithUnclaimedRewards removes addresses also when rewards are still unclaimed
If we redeem, for example, only the nativeToken rewards but not the ETH rewards, the address of the beneificiary will be removed from accountsWithUnclaimedRewards. But the ETH rewards are still there to redeem.

This occurs naturally if we use the `Redeem.contributionRewardRedeem` function when the ETH balance of the DAO avatar is not high enough to reward the user.

### PR DESCRIPTION
**Pull request by ben-kaufman**
Fix #206

### CODE REVIEW

File: src/domain/proposal.ts [hunk 1]
```diff
@@ -204,6 +205,8 @@ export function updateCRProposal(
   setSchemeName(proposal.scheme, 'ContributionReward');
   getProposalIPFSData(proposal);
 
+  addRedeemableRewardOwner(proposalId, beneficiary);
```
Comment (orenyodfat) [inline src/domain/proposal.ts]: why is this needed ?

File: src/domain/reward.ts [hunk 2]
```diff
@@ -81,6 +81,32 @@ export function reputationRedemption(proposalId: Bytes, beneficiary: Address, ti
 }
 
 function shouldRemoveAccountFromUnclaimed(reward: GPReward): boolean {
+  let proposal = getProposal(reward.proposal);
+  if (proposal.contributionReward !== null) {
+    let crProposal = store.get(
```
Comment (orenyodfat) [inline src/domain/reward.ts]: please use ContributionRewardProposal.load

File: src/domain/reward.ts [hunk 3]
```diff
@@ -81,6 +81,32 @@ export function reputationRedemption(proposalId: Bytes, beneficiary: Address, ti
 }
 
 function shouldRemoveAccountFromUnclaimed(reward: GPReward): boolean {
+  let proposal = getProposal(reward.proposal);
```
Comment (orenyodfat) [inline src/domain/reward.ts]: why need this ? 
use ContributionRewardProposal.load(pid) and check if its not null.

**Comment by orenyodfat** [comment]
please add tests to cover this issue.

File: src/domain/proposal.ts (hunk 1 above)
Comment (ben-kaufman) [reply src/domain/proposal.ts]: To make sure the beneficiary is listed as a redeemer

File: src/domain/reward.ts [hunk 4]
```diff
@@ -81,6 +81,28 @@ export function reputationRedemption(proposalId: Bytes, beneficiary: Address, ti
 }
 
 function shouldRemoveAccountFromUnclaimed(reward: GPReward): boolean {
+  let proposal = ContributionRewardProposal.load(reward.proposal);
+  if (proposal !== null) {
+    if (equalsBytes(proposal.beneficiary, reward.beneficiary)) {
```
Comment (orenyodfat) [inline src/domain/reward.ts]: does this cover by any test?

**Pull request merged**